src/options.ts

```diff
@@ -27,7 +31,7 @@ export interface IOptions extends ValidatorOptions {
    * optionally defined in order to override the default storage used
    * to parse decorator metadata.
    */
-  classValidatorMetadataStorage: MetadataStorage
+  classValidatorMetadataStorage: IStorage
```

Comment:
This option should stay compatible with class-validator, otherwise users will have to do type coercion like `classValidatorMetadataStorage: myCustomStorage as IStorage`. So let's keep `classValidatorMetadataStorage` as of type `MetadataStorage` and handle any type coercion internally if necessary.

WIth this change the `IStorage` type probably won't be needed anymore?